pkg/opensearch/ppl_response_parser.go

```diff
@@ -100,14 +116,22 @@ func (rp *pplResponseParser) parseLogs(queryRes *backend.DataResponse, configure
 		docs[rowIdx] = doc
 	}
 
-	sortedPropNames := sortPropNames(propNames, []string{configuredFields.TimeField, configuredFields.LogMessageField})
+	fieldsToGoInFront := []string{}
+	if isLogsQuery {
+		fieldsToGoInFront = append(fieldsToGoInFront, configuredFields.TimeField, configuredFields.LogMessageField)
+	}
+	sortedPropNames := sortPropNames(propNames, fieldsToGoInFront)
 	fields := processDocsToDataFrameFields(docs, sortedPropNames)
 
 	frame := data.NewFrame("", fields...)
 	if frame.Meta == nil {
 		frame.Meta = &data.FrameMeta{}
 	}
+
 	frame.Meta.PreferredVisualization = data.VisTypeLogs
+	if !isLogsQuery {
+		frame.Meta.PreferredVisualization = data.VisTypeTable
```

Comment:
technically this isn't in the frontend code (it doesn't set it) but it made sense as an improvement to me